From c243e2904de4e1ab12ffe294ab2a5a89c089a7aa Mon Sep 17 00:00:00 2001
From: Hien Huynh <hien.huynh.px@renesas.com>
Date: Mon, 12 Oct 2020 16:52:47 +0700
Subject: [PATCH 17/18] driver: rzg: swdt: increase WDT count when ECC enabled

When enable ECC, access to dram have lower speed, sometimes Watchdog
timer get overflow when transfering Uboot binary from flash to DRAM.
This commit increase WDT counter to it's maximum value.

Signed-off-by: Binh Nguyen <binh.nguyen.uw@renesas.com>
Signed-off-by: Hien Huynh <hien.huynh.px@renesas.com>
---
 drivers/renesas/rzg/watchdog/swdt.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/renesas/rzg/watchdog/swdt.c b/drivers/renesas/rzg/watchdog/swdt.c
index b694aa3..619910e 100644
--- a/drivers/renesas/rzg/watchdog/swdt.c
+++ b/drivers/renesas/rzg/watchdog/swdt.c
@@ -49,9 +49,15 @@ extern void gicd_set_icenabler(uintptr_t base, unsigned int id);
 #define WTCSRA_INIT_DATA		(WTCSRA_UPPER_BYTE + 0x0FU)
 #define WTCSRB_INIT_DATA		(WTCSRB_UPPER_BYTE + 0x21U)
 
+#if (RZG_DRAM_ECC == 1)
+#define WTCNT_COUNT_8p13k		(0U)
+#define WTCNT_COUNT_8p22k		(0U)
+#define WTCNT_COUNT_7p81k		(0U)
+#else /* (RZG_DRAM_ECC == 1) */
 #define WTCNT_COUNT_8p13k		(0x10000U - 40687U)
 #define WTCNT_COUNT_8p22k		(0x10000U - 41115U)
 #define WTCNT_COUNT_7p81k		(0x10000U - 39062U)
+#endif /* (RZG_DRAM_ECC == 1) */
 
 #define WTCSRA_CKS_DIV16		(0x00000002U)
 
-- 
2.7.4

